cli/compose/convert/service.go

```diff
@@ -476,3 +488,20 @@ func convertDNSConfig(DNS []string, DNSSearch []string) (*swarm.DNSConfig, error
 	}
 	return nil, nil
 }
+
+func convertCredentialSpec(source string) (*swarm.CredentialSpec, error) {
+	if source == "" {
+		return nil, nil
+	}
+
+	credentialSpec := swarm.CredentialSpec{}
+	switch {
+	case strings.HasPrefix(source, "file://"):
```

Comment:
Conversation from the PR https://github.com/moby/moby/pull/32393#discussion_r114618977

I would really prefer we don't do this string parsing in the yaml format.
It's fine in the CLI but in the yaml format it's not saving anybody any time to do this in the yaml format.
It's better to have a well defined structure, the API even models this.